add page on attestations to pos section (#7155)
ethereum/ethereum-org-website: src/content/developers/docs/consensus-mechanisms/pos/attestations/index.md

---
title: Attestations

> description: A description of attestations on PoS Ethereum.

Reviewer: ```suggestion
description: A description of attestations on proof-of-stake Ethereum.
```

lang: en
sidebar: true
---

A validator is expected to create, sign and broadcast an attestation during every epoch. This page outlines what these attestations look like and how they are processed and communciated between consensus clients.

> ## What is an attestation? {#what-is-an-attestation}

Reviewer: Can we add pre-requisites here

- `data`: details relating to the attestation, as defined below
- `signature`: a BLS signature that aggregates the signatures of individual validators

The first task for an attesting validator is to build the `data`. The `data` contains the following information:

- `slot`: The slot number that the attestation refers to
- `index`: A number that identifies which committee the validator belongs to in a given slot
- `beacon_block_root`: Root hash of the block the validator sees at the head of the chain (the result of applying the fork-choice algorithm)
- `source`: Part of the finality vote indicating what the validators see as the most recent justified block
- `target`: Part of the finality vote indicating what the validators see as the first block in the current epoch

Once the `data` is built, the validator can flip the bit in `aggregation_bits` corresponding to their own validator index from 0 to 1 to show that they participated.

Finally, the validator signs the attestation and broadcasts it to the network.

> ### Aggregated Attestation {#aggregated-attestation}

Reviewer: ```suggestion
### Aggregated attestation {#aggregated-attestation}
```

There is a substantial overhead associated with passing this data around the network for every validator. Therefore, the attestations from individual validators are aggregated within subnets before being broadcast more widely. This includes aggregating signatures together so that an attestation that gets broadcast includes the consensus `data` and a single signature formed by combining the signatures of all the validators that agree with that `data`. This can be checked using `aggregation_bits` because this provides the index of each validator in their committee (whose ID is provided in `data`) which can be used to query individual signatures.

In each epoch a validator in each subnet is selected to be the `aggregator`. The aggregator collects all the attestations it hears about over the gossip network that have equivalent `data` to its own. The sender of each matching attestation is recorded in the `aggregation_bits`. The aggregator then broadcasts the attestation aggregate to the wider network.

When a validator is selected to be a block proposer they package aggregate attestations from the subnets up to the latest slot in the new block.

> ### Attestation Inclusion Lifecycle {#attestation-inclusion-lifecycle}

Reviewer: Please use sentence case for titles :-)

```suggestion
### Attestation inclusion lifecycle {#attestation-inclusion-lifecycle}
```
